alphadia/workflow/peptidecentric.py

```diff
@@ -1102,22 +1102,7 @@ def _build_candidate_speclib_flat(
     psm_df: pd.DataFrame,
     fragment_types: typing.List[str] = ["b", "y"],
     max_charge: int = 2,
-    optional_columns: typing.List[str] = [
-        "proba",
-        "score",
-        "qval",
-        "channel",
-        "rt_library",
-        "mz_library",
-        "mobility_library",
-        "genes",
-        "proteins",
-        "decoy",
-        "mods",
-        "mod_sites",
-        "sequence",
-        "charge",
-    ],
+    optional_columns: typing.Union[typing.List[str], None] = None,
```

Comment:
(nit) we should move the whole codebase from `typing.Union[typing.List[str], None]` to ` list | None`. (not in this PR :-))

Reply:
hahaha, what's the idea behind this? I guess my typing knowledge is deprecated :D